### PR TITLE
Add parallel_cross_entropy multi cards test

### PR DESCRIPTION
**Pull request by wangxn12138**
1.inbubate的稳定性测试跳过，因为API与paddle.nn.CrossEntropyLoss对应的incubate API相同（fluid.layers.softmax_with_cross_entropy），因此incubate API的稳定性测试已经在CrossEntropyLoss测试中被完成，不再重复测试
2.kernel不支持bf16数据类型，运行结果：
image

测试case:
logits shape:[1,46256]
develop测试误差：
正确性误差：
fp16，2卡，动态图，前向运行结果：
image
fp16，2卡，动态图，反向运行结果：
image
fp16，2卡，静态图，前向运行结果：
image
fp16，2卡，静态图，反向运行结果：
image
fp16，4卡，动态图，前向运行结果：
image
fp16，4卡，静态图，前向运行结果：
image
fp16，8卡，动态图，前向运行结果：
image
fp16，8卡，静态图，前向运行结果：
image
稳定性误差：
fp32，4卡，动态图，反向稳定性运行结果：
image
fp32，4卡，静态图，反向稳定性运行结果：
image

incubate测试误差：
正确性误差：
对比2卡测试，fp32，动态图，反向测试结果：
image
对比2卡测试，fp32，静态图，反向测试结果：
image
对比4卡测试，fp32，动态图，反向测试结果：
image
对比4卡测试，fp32，静态图，反向测试结果：
image
对比8卡测试，fp32，动态图，反向测试结果：
image
对比8卡测试，fp32，静态图，反向测试结果：
image
对比2卡测试，fp16，动态图，前向测试结果：
image
对比2卡测试，fp16，静态图，前向测试结果：
image
对比4卡测试，fp16，动态图，前向测试结果：
image
对比4卡测试，fp16，静态图，前向测试结果：
image
对比8卡测试，fp16，动态图，前向测试结果：
image
对比8卡测试，fp16，静态图，前向测试结果：
image

### CODE REVIEW

File: test_parallel_cross_entropy/get_torch_result.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import torch.distributed as torch_dist
+import init_config_class
+import prepare_data
+import sys
+sys.path.append("..")
+from utils import TOLERANCE, convert_dtype_to_torch_type
+
+class TestTorch(init_config_class.InitConfigClass):
+    def __init__(self, device, np_input_dir="./inputs_case1.npz", dtype="float32", torch_dir="1_torch_out_float32.npz"):
+        self._init_params(np_input_dir, dtype)
```
Comment (xiaoguoguo626807) [inline test_parallel_cross_entropy/get_torch_result.py:12]: 哪个文件生成的数据，torch 和 develop都是读取数据计算的？

**Review by xiaoguoguo626807 — COMMENTED** [review]
LGTM
